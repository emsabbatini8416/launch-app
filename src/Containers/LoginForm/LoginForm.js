import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { useForm } from "react-hook-form";

import LabelInput from './../../Components/LabelInput'

import { loginToken } from './../../Actions/Users'

import './LoginForm.scss';

export function LoginForm() {
  const { register, handleSubmit, errors } = useForm();
  const dispatch = useDispatch();
  const history = useHistory();

  const onSubmit = data => {
    dispatch(loginToken(data));
  };

  return (
    <div className="login-form">
      <div className="login-form_container">
        <div className="row justify-content-center align-items-center">
          <div className="col-md-6">
            <div className="col-md-12">
              <form className="form" onSubmit={handleSubmit(onSubmit)}>
                <h3>Login</h3>
                <div className="form-group">
                  <LabelInput 
                    label={'Username'} 
                    type={'text'} 
                    name={'username'}   
                    register={register({
                      required: "You must specify an username"
                    })} 
                    errors={errors}
                  />
                </div>
                <div className="form-group">  
                  <LabelInput 
                    label={'Password'} 
                    type={'password'} 
                    name={'password'}   
                    register={register({
                      required: "You must specify a password"
                    })} 
                    errors={errors}
                  />
                </div>
                <div className="form-group">
                    <input type="submit" name="submit" className="btn login-form_container--btn" value="Submit" />
                    <input type="button" name="signup" className="btn" onClick={() => history.push('/signup')} value="Sign Up" />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default LoginForm;