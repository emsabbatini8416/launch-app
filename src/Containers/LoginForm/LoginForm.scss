.login-form {
    display: flex;
    flex-direction: column;
    align-items: center;
    &_container {
        max-width: 600px;
        min-width: 600px;
        height: 320px;
        border: 1px solid #9C9C9C;
        background-color: #EAEAEA;
        h3, label {
            color: #143e66; 
        }
        &--btn {
            color: white !important;
            background-color: #143e66 !important;
        }
    }
}
