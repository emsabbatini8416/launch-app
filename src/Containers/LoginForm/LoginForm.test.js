import React from 'react';
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import LoginForm from './';

const mockStore = configureMockStore([thunk]);
const mockState = {}

describe('LoginForm component tests', () => {
    let store = mockStore(mockState)

    const wrapper = mount(
        <Provider store={store}>
            <LoginForm />
        </Provider>);

    it('should have a LogIn btn component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should have a LogIn btn component', () => {
        //There should be only one button
        expect(wrapper.find('input#btnLogIn')).toHaveLength(1);

        //Button should be of type button
        expect(wrapper.find('input#btnLogIn')
        .prop('type'))
        .toEqual('submit');

        //Button should have matching text
        expect(wrapper.find('input#btnLogIn')
        .prop('value'))
        .toEqual('LogIn');
    });

    it('should have a SignUp btn component', ()=> {
        //There should be only one button
        expect(wrapper.find('input#btnSignUp')).toHaveLength(1);

        //Button should be of type button
        expect(wrapper.find('input#btnSignUp')
        .prop('type'))
        .toEqual('button');

        //Button should have matching text
        expect(wrapper.find('input#btnSignUp')
        .prop('value'))
        .toEqual('SignUp');
    });

    it('should have input for email and password', ()=> {
        //Email and password input field should be present
        expect(wrapper.find('LabelInput')).toHaveLength(2);
    });
});
