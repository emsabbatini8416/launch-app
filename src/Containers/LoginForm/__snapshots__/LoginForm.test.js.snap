// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component tests should have a LogIn btn component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LoginForm>
    <div
      className="login-form"
    >
      <div
        className="login-form_container"
      >
        <div
          className="row justify-content-center align-items-center"
        >
          <div
            className="col-md-6"
          >
            <div
              className="col-md-12"
            >
              <form
                className="form"
                onSubmit={[Function]}
              >
                <h3>
                  Login
                </h3>
                <div
                  className="form-group"
                >
                  <LabelInput
                    errors={Object {}}
                    label="Username"
                    name="username"
                    register={[Function]}
                    type="text"
                  >
                    <label
                      htmlFor="username"
                    >
                      Username
                    </label>
                    <br />
                    <input
                      className="form-control"
                      name="username"
                      type="text"
                    />
                  </LabelInput>
                </div>
                <div
                  className="form-group"
                >
                  <LabelInput
                    errors={Object {}}
                    label="Password"
                    name="password"
                    register={[Function]}
                    type="password"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <br />
                    <input
                      className="form-control"
                      name="password"
                      type="password"
                    />
                  </LabelInput>
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="btn login-form_container--btn"
                    id="btnLogIn"
                    name="submit"
                    type="submit"
                    value="LogIn"
                  />
                  <input
                    className="btn"
                    id="btnSignUp"
                    name="signup"
                    onClick={[Function]}
                    type="button"
                    value="SignUp"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </LoginForm>
</Provider>
`;
